pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -30,19 +33,18 @@ func NewAnnotationHistorian(annotations annotations.Repository, dashboards dashb
 	}
 }
 
-func (h *AnnotationStateHistorian) RecordState(ctx context.Context, rule *ngmodels.AlertRule, labels data.Labels, evaluatedAt time.Time, currentData, previousData state.InstanceStateAndReason) {
+func (h *AnnotationStateHistorian) RecordState(ctx context.Context, rule *ngmodels.AlertRule, currentState *state.State, evaluatedAt time.Time, currentData, previousData state.InstanceStateAndReason) {
```

Comment:
I think we can drop `currentData` as it's a subset of `currentState`. This is also consistent with what my current PR also does: https://github.com/grafana/grafana/blob/730223909ef812b6944e4f6b7c9236c9a955f869/pkg/services/ngalert/state/historian/annotation.go#L34 (stateTransition is currentState + previousData embedded). Don't bother holding this PR on that change, though.